server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceMetrics.java

```diff
@@ -112,6 +130,36 @@ public DesiredBalanceMetrics(MeterRegistry meterRegistry) {
             "unit",
             this::getDesiredBalanceNodeShardCountMetrics
         );
+        meterRegistry.registerDoublesGauge(
+            CURRENT_NODE_WRITE_LOAD_METRIC_NAME,
+            "The current write load of nodes",
+            "threads",
+            this::getCurrentNodeWriteLoadMetrics
+        );
+        meterRegistry.registerDoublesGauge(
+            CURRENT_NODE_DISK_USAGE_METRIC_NAME,
+            "The current disk usage of nodes",
+            "bytes",
+            this::getCurrentNodeDiskUsageMetrics
+        );
+        meterRegistry.registerLongsGauge(
+            CURRENT_NODE_SHARD_COUNT_METRIC_NAME,
+            "The current shard count of nodes",
+            "unit",
+            this::getCurrentNodeShardCountMetrics
+        );
+        meterRegistry.registerDoublesGauge(
+            CURRENT_NODE_FORECASTED_DISK_USAGE_METRIC_NAME,
+            "The current forecasted disk usage of nodes",
+            "bytes",
+            this::getCurrentNodeForecastedDiskUsageMetrics
+        );
+        meterRegistry.registerLongsGauge(
+            CURRENT_NODE_UNDESIRED_SHARD_COUNT_METRIC_NAME,
+            "The current undesired shard count of nodes",
+            "unit",
+            this::getCurrentNodeUndesiredShardCountMetrics
```

Comment:
Am I right in thinking the existing undesired metric `es.allocator.desired_balance.allocations.undesired.current` is a cluster-wide metric, and this one is the same measure, but broken down by node?

Reply:
yes, that's my understanding as well.